### PR TITLE
Temporary page to create users in db

### PR DESCRIPTION
**Pull request by RobinBurri**
This branch has a new page in the front : /tmp
You can create a new Player in the database, get existing players and delete them.

This feature is TEMPORARY and will be deleted later when it won't be needed anymore.

* add `/api` for each URL

### CODE REVIEW

**Review by eprei — APPROVED** [review]
Great job Robin. This PR not only adds the functionality to add users to the database but also serves as a tutorial on how the whole process works from frontend, through the backend and up to the database. Thanks for this work.
I just need to see in depth the TempForm.tsx and TempPlayerList.tsx files but as it is a temporary feature, I think I have already seen its logic and for me this PR is already valid. I will keep looking at these two files.

**Review by shaolin-peanut — APPROVED** [review]
Looks good to me. C'est temporaire donc je vais passer 2 jours a faire une review in-depth, mais ça fait des éléments cool a étudier comme example pour comment implémenter certains trucs dans notre infrastructure

File: backend/src/player/player.controller.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import {
+    Controller,
+    Get,
+    Post,
+    Body,
+    Patch,
```
Comment (t-h2o) [inline backend/src/player/player.controller.ts:6]: Patch is not use
```suggestion
```

File: backend/src/player/player.controller.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    @UsePipes(ValidationPipe)
+    async create(@Body() createPlayerDto: CreatePlayerDto) {
+        console.log(createPlayerDto)
+        const player = this.playerService.create(createPlayerDto)
+        return player
+    }
+
+    @Get()
+    async findAll() {
+        const player = await this.playerService.findAll()
```
Comment (t-h2o) [inline backend/src/player/player.controller.ts:29]: Should we be more explicit with types ?
```suggestion
    async findAll(): Promise<Player[]> {
        const player: Player[] = await this.playerService.findAll()
```
without forget to import `Player` with
```ts
import { Player } from 'src/typeorm/Player'
```

File: backend/src/player/player.service.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    constructor(
+        @InjectRepository(Player)
+        private readonly playerRepository: Repository<Player>
+    ) {}
+    create(createPlayerDto: CreatePlayerDto) {
+        const newPlayer = this.playerRepository.create(createPlayerDto)
+        return this.playerRepository.save(newPlayer)
+    }
+
+    findAll() {
```
Comment (t-h2o) [inline backend/src/player/player.service.ts:18]: Add type ?
```suggestion
    findAll(): Promise<Player[]> {
```

File: backend/src/player/player.service.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        return this.playerRepository.find()
+    }
+
+    findOne(id: number) {
```
Comment (t-h2o) [inline backend/src/player/player.service.ts:22]: Add type ?
```suggestion
    findOne(id: number):string {
```

File: frontend/src/components/tempLogin/TempPlayerLi.module.css
Comment (fire-poy) [inline frontend/src/components/tempLogin/TempPlayerLi.module.css:1]: Into class .li change
   background-color: var(--color-dark-green);
for
   background-color: var(--color-black);

File: trans.session.sql
[the remark this answers is not in the document]
Comment (RobinBurri) [reply trans.session.sql:1]: It's a file created by an vsc extension to connect the db.
deleted fc90e5d8c1b96f41715a1c890ced972b772bb3b4

File: backend/src/player/player.controller.ts (hunk 1 above)
Comment (RobinBurri) [reply backend/src/player/player.controller.ts]: done
7c43062a2d4bf0bbfeda2586c5934539f881b940

File: backend/src/player/player.controller.ts (hunk 2 above)
Comment (RobinBurri) [reply backend/src/player/player.controller.ts:28]: I have been reading some articles about that:
https://dev.to/pffigueiredo/typescript-return-types-separating-fact-from-fiction-4266
https://www.totaltypescript.com/tips/dont-use-return-types-unless

It's a choice, I think that in this case it's pretty clear:
async findAll() in class Player return a Promise <Player[]>

File: backend/src/player/player.service.ts (hunk 3 above)
Comment (RobinBurri) [reply backend/src/player/player.service.ts:18]: same as above, the function is simple enough, no return type needed, imo.

File: backend/src/player/player.service.ts (hunk 4 above)
Comment (RobinBurri) [reply backend/src/player/player.service.ts:22]: This one has been created by nest cli, `This action returns a #${id} player` is a placeholder that is needed to change.

File: frontend/src/components/tempLogin/TempPlayerLi.module.css
Comment (RobinBurri) [reply frontend/src/components/tempLogin/TempPlayerLi.module.css:1]: /tmp page is going to be deleted later, so the styling is not important.

File: backend/src/player/player.controller.ts (hunk 2 above)
Comment (t-h2o) [reply backend/src/player/player.controller.ts:28]: Interesting, thank you for the resources

File: backend/src/player/player.service.ts (hunk 4 above)
Comment (t-h2o) [reply backend/src/player/player.service.ts:22]: OK

File: backend/src/player/player.service.ts (hunk 3 above)
Comment (t-h2o) [reply backend/src/player/player.service.ts:18]: Good

File: trans.session.sql
[the remark this answers is not in the document]
Comment (t-h2o) [reply trans.session.sql:1]: Good

**Review by t-h2o — APPROVED** [review]

**Review by fire-poy — APPROVED** [review]

**Pull request merged**